jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -728,7 +728,7 @@ public void onCompleted()
         if (LOG.isDebugEnabled())
             LOG.debug("onCompleted for {} written app={} net={}", apiRequest.getRequestURI(), getHttpOutput().getWritten(), getBytesWritten());
 
-        if (getServer().getRequestLog() instanceof CustomRequestLog)
+        if (getServer().getRequestLog() instanceof CustomRequestLog customRequestLog && customRequestLog.isLogDetailRequired())
```

Comment:
You could wrap this up in a static method that reads a lot better
```suggestion
        if (CustomRequestLog.isLogDetailRequired(getServer()))
```

Reply:
done